File: backend/src/main/java/com/backend/backend/models/Product.java
package com.backend.backend.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Document(collection = "products")
public class Product {
    @Id
    private String productId;

    private String name;
    private String description;
    private Double price;
    private String imageUrl;
    private int rating;
    private int no_of_ratings;
    private Double basePrice;

    @DBRef
    private Category category;
    @DBRef // Reference to another document in MongoDB
    private User designer;
    @DBRef
    private Customization customization;

    @CreatedDate
    private LocalDateTime createdAt;
    @LastModifiedDate
    private LocalDateTime updatedAt;

}

File: frontend/src/models/product.js
// Product.js
export class Product {
  constructor(
    productId,
    name,
    description,
    price,
    basePrice,
    imageUrl,
    customization,
    creator,
    category, 
    rating
  ) {
    this.productId = productId; // Unique ID for the product
    this.name = name; // Product name
    this.description = description; // Product description
    this.price = price; // Product price
    this.basePrice = basePrice; // Base price
    this.imageUrl = imageUrl; // Image URL of the product
    this.customization = customization;
    this.creator = creator; // Creator of the product 
    this.category = category;
    this.rating = rating; // Rating of the product
  }
}
